### PR TITLE
Added a Tutorial page on debugging apps with VS Code docker extension

### PR DESCRIPTION
**Pull request by gunamata**
Signed-off-by: Gunasekhar Matamalam <gunasekhar.matamalam@suse.com>

### CODE REVIEW

**Review by evertonlperes — APPROVED** [review]
LGTM - Works nicely on macOS

**Review by jtravee — CHANGES_REQUESTED** [review]

File: docs/tutorials/vs-code-docker.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+title: Debugging a Container App with VS Code Docker extension
+---
+
+The VS Code Docker extension makes it easy to build, manage, debug and deploy containerized applications in Visual Studio Code.
+
+### Steps to debug a sample application running within a container.
```
[the remark this answers is not in the document]
Comment (gunamata) [reply docs/tutorials/vs-code-docker.md]: Modified per your suggestion.

File: docs/tutorials/vs-code-docker.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+[VS Code Docker]: https://code.visualstudio.com/docs/containers/overview
+
+3. Install the VS Code Docker extension from the marketplace. 
+
+![](/img/vscodedocker/vscode-docker-marketplace.png) 
+
+4. You can use the samples provided at this Github repository, https://github.com/bwateratmsft/samples. Clone this repository and open `expressapp` folder in your VS Code session. 
+
+6. Open the command palette (Ctrl+Shift+P, F1, or Cmd+Shift+P) and run "Add Docker Files to Workspace". Since this is an Express app, Select `Node.js` as Application Platform,  `3000` or any other available port as `port`. As it's a simple example, select `No` for the `Include optional Docker Compose files` prompt. This step adds a `Dockerfile` and `Launch Configuration` required to debug the application.
```
[the remark this answers is not in the document]
Comment (gunamata) [reply docs/tutorials/vs-code-docker.md]: Modified per your suggestion.

File: docs/tutorials/vs-code-docker.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+![](/img/vscodedocker/vscode-docker-debug-configuration.png)
+
+![](/img/vscodedocker/vscode-docker-debug-breakpoint-hit.png)
+
+9. Sometimes, the application might not break at the set breakpoint on the first run as the debugging process may not have started. In this case, refresh the browser to trigger the execution again to hit the breakpoint. You can also get around this behavior by setting the property `inspectMode: 'break'` in `task.json` file to prevent the app from running until the debugger attaches.
+
+10. On some machines, the firewall settings might prevent the debugging process from establishing a connection between the host and the container processes. In this case, you can add a firewall rule to allow communication between the VM where the container is running and the host where you have VS Code session running. On Windows, you can add a firewall rule by running below command from a privileged powershell.
```
[the remark this answers is not in the document]
Comment (gunamata) [reply docs/tutorials/vs-code-docker.md]: Modified per your suggestion.

**Review by jtravee — APPROVED** [review]

**Pull request merged**